### PR TITLE
Fix with gcc 7.3

### PR DESCRIPTION
**Pull request by pkrusche**
This fixes the gcc 4.8.x regex error and unit test failures with gcc 7.3. Also, the Docker build is fixed.

### CODE REVIEW

**Comment by pkrusche** [comment]
See discussion in #66

**Comment by dpryan79** [comment]
Is there an ETA for getting this merged and a new version tagged? There are a number of bioconda users running into issues due to this.

File: Dockerfile [hunk 1]
```diff
@@ -14,17 +15,17 @@ RUN apt-get install python2.7 python2.7-dev python \
...
                     -y
-RUN apt-get install software-properties-common python-software-properties -y
+RUN apt-get install software-properties-common -y
 RUN apt-get clean -y
 
 RUN pip install bx-python
 
-RUN echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | debconf-set-selections && \
-    add-apt-repository -y ppa:webupd8team/java && \
+RUN echo oracle-java11-installer shared/accepted-oracle-license-v1-2 select true | debconf-set-selections && \
```
Comment (phageghost) [inline Dockerfile]: Internally we've switched to openjdk8 without problems. Also solves the license issues with Oracle Java which would still exist with this fix.

File: Dockerfile.ubuntu-with-tests [hunk 2]
```diff
@@ -14,17 +15,17 @@ RUN apt-get install python2.7 python2.7-dev python \
...
                     -y
-RUN apt-get install software-properties-common python-software-properties -y
+RUN apt-get install software-properties-common -y
 RUN apt-get clean -y
 
 RUN pip install bx-python
 
-RUN echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | debconf-set-selections && \
-    add-apt-repository -y ppa:webupd8team/java && \
+RUN echo oracle-java11-installer shared/accepted-oracle-license-v1-2 select true | debconf-set-selections && \
```
Comment (phageghost) [inline Dockerfile.ubuntu-with-tests]: See above

**Pull request merged**